### PR TITLE
Revamp tutorial writing pages

### PR DESCRIPTION
**Pull request by ganicke**
I've taken the tutorial page and split it into multiple topics. The single page is too crammed up and is difficult for guidance. It now goes like this...

## Writing tutorials

* [Basics](#) - The basics of writing a tutorial
* [Activities](#) - Creating an activity based tutorial
* [Control options](#) - Additional options and blocks to control tutorial behavior
* [Multi-language](#) - Tutorials for both Blocks and Code
* [Adding Resources](#) - Including a resource section in a tutorial
* [Publish](#) - How to publish a tutorial

re: https://github.com/microsoft/pxt-minecraft/issues/2017

### CODE REVIEW

File: docs/writing-docs/tutorials/activities.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+### Activity step structure
+
+The activity tutorial structure uses a two-level layout - activity and steps, next activity.
+
+```
+### @activities 1
+
+## Introduction
+    ### Introduction step
```
Comment (pelikhan) [inline docs/writing-docs/tutorials/activities.md:40]: do not indent

**Review by pelikhan — APPROVED** [review]

File: docs/writing-docs/tutorials/basics.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+To add a special behavior to a step, use a step modifier:
+
+### fullscreen
+
+If you want to include a dramatic introduction or make certain that a special message is seen, you can use the ``@fullscreen`` tag. The section is displayed in an overlay window on top of the tutorial screen and isn't shown in the caption as part of the tutorial flow. You include it in your tutorial like this:
+
+```markdown
+# Flash-a-rama
+
+## It's time to code! @fullscreen
```
Comment (shakao) [inline docs/writing-docs/tutorials/basics.md:121]: so "fullscreen" auto-pops the hint and "unplugged" displays the full screen modal (and auto-advances to the next step).... maybe we should rename these

**Review by Jaqster — APPROVED** [review]
Looks great!  Thanks Galen!

File: docs/writing-docs/user-tutorials.md [hunk 3]
```diff
@@ -74,6 +74,16 @@ In context tutorials, sometime referred to as "recipes", are tutorials that are
 
 Click on the ``lab`` icon in the **Explorer** view to open any markdown file (``.md``) as a tutorial in a new tab.
 
+### ~ alert
+
+#### Tutorial caching
+
+MakeCode uses a local caching policy for tutorials to reduce interaction with website services. On first use, tutorial content is retrieved from a MakeCode website and then reused from the local cache when a tutorial is run another time. A requested tutorial will refresh from the website when its cache retention period expires.
+
+This caching policy can present a problem if you're developing a tutorial and want to review the recent changes. When you run the tutorial to check your changes, they might not appear and you only see content you viewed the first time. In order to see and test new changes you've published for your tutorial, it's recommended that your view them in a **new anonyomous / incognito** browser window.
```
Comment (shakao) [inline docs/writing-docs/user-tutorials.md:83]: not quite sure where to put this, but if they are writing Github tutorials specifically, they will need to create a release using the MakeCode Github integration to force-clear the cloud cache (documentation here: https://arcade.makecode.com/github/release)

**Review by shakao — APPROVED** [review]

File: docs/writing-docs/tutorials/control-options.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    . . . . . . . . 
+
+    `, SpriteKind.Player)
+game.onUpdateInterval(1000, function () {
+	
+})
+```
+````
+
+You can include **1** template anywhere in the tutorial.
```
[the remark this answers is not in the document]
Comment (ganicke) [reply docs/writing-docs/tutorials/control-options.md]: Yea. It's best not to break the flow sequence of the steps too. Note added.

File: docs/writing-docs/user-tutorials.md (hunk 3 above)
Comment (ganicke) [reply docs/writing-docs/user-tutorials.md]: See if my in note here https://github.com/microsoft/pxt/pull/7481/commits/f150b055f379e645dbf9ff5d468e586cadb37717 is legit.

Comment (shakao) [reply docs/writing-docs/user-tutorials.md]: looks awesome, thanks galen!!

**Pull request merged**